### DISCUSSION TITLE
Radio button works incorrectly when patch from value

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [taiga-family/taiga-ui#9801] [Q&A] [answered]
I am attaching a vuideo but it is some kind of tick missing because when i path from value (after 2 seconds as you can see in source code) radio button value changes but animation is not played until the end. It is enough for me click anywhere on the screen and animation is completed. Any ideas why it could happen?

https://github.com/user-attachments/assets/f3bc5a15-96ed-4c5d-be8f-348adc532463

**Comment by vytautas-pranskunas-** [comment]
I have done few thing during my investigation. I had extracted code with form to seperate component and added this component over existing radio buttons. It looks and works different. First off all ther is no data-appearance="primary" on radiobuttons inside of extracted component but there on screen radio buttons. Second thing that in extracted component there is white big circle of ::before. Any ideas why? But animation is capable to play fully

code of extracted component:
`
import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, inject, OnInit } from '@angular/core';
import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
import { TitlePrintOption } from '@api';
import { TranslateModule } from '@ngx-translate/core';
import { TuiTextfield } from '@taiga-ui/core';
import { TuiRadio } from '@taiga-ui/kit';

@Component({
    selector: 'pa-test-component',
    standalone: true,
    templateUrl: './test-component.component.html',
    imports: [CommonModule, TranslateModule, ReactiveFormsModule, TuiTextfield, TuiRadio],
    changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TestComponent implements OnInit {
    private readonly formBuilder = inject(FormBuilder);

    form = this.formBuilder.group({
        printDepartmentOption: [TitlePrintOption.DoNotPrint],
    });

    ngOnInit(): void {
        setTimeout(() => {
            this.form.patchValue({ printDepartmentOption: TitlePrintOption.Print });
        }, 2000);
    }
}

`

`
<form [formGroup]="form">
  <div class="column-layout">
      
      <div class="column">
          <div>
              <label class="radio">
                  <input formControlName="printDepartmentOption" tuiRadio type="radio" value="DoNotPrint" />
                  {{ 'appeal-edit.doNotPrintDepartmentName' | translate }}
              </label>
          </div>
          <div>
              <label class="radio">
                  <input formControlName="printDepartmentOption" tuiRadio type="radio" value="Print" />
                  {{ 'appeal-edit.printDepartmentName' | translate }}
              </label>
          </div>
          <div>
              <label class="radio">
                  <input formControlName="printDepartmentOption" tuiRadio type="radio" value="PrintOther" />
                  {{ 'appeal-edit.printDepartmentNameCustom' | translate }}
              </label>
          </div>
          <div>
            <tui-textfield>
                <input tuiTextfield formControlName="otherDepartmentName" />
            </tui-textfield>
        </div>
      </div>

      
      <div class="column"></div>
  </div>
</form>

`

https://github.com/user-attachments/assets/dd5ed955-085d-4154-8dab-bc398b2c517d

**Reply by splincode** [reply]
Could you isolate your example on Stackblitz?

**Reply by vytautas-pranskunas-** [reply]
Yes, i was able to reproduce it.. Issue with not fully selected radio button: https://stackblitz.com/edit/angular-buf4yk?file=src%2Fapp%2Fapp.template.html,src%2Fapp%2Fapp.component.ts&hideExplorer=1

however i still wonder why when it is in seperate component the look of radio is different - i was not able to reproduce that part :(

**Reply by vytautas-pranskunas-** [reply]
p.s. also can you advice why textField is not disabled at first load and becomes disabled only after subsecuent clicking of radio buttons?

**Reply by waterplea** [reply]
You have `OnPush`. Either add `this.cdr.markForCheck();` or remove `OnPush`. It is expected not to pick up on changes when you use `setTimeout` with `OnPush` change detection strategy, that's on Angular, not on Taiga UI.

**Reply by vytautas-pranskunas-** [reply]
i stongly do not agree, because it is not setatimout. SetTimeout only mimics api call. Pls show me in docs where it is written that form.patch should be used with detectChanges after api data fetch when using onPush. Plus it is clearly visible that it changes checked radio just animation is not player well, plus if you try it with normal radio or any other component it will work

**Reply by vytautas-pranskunas-** [reply]
P.s. in real ptoject i use state managment system which does markForCheck while syncing with form state but problem persists.

**Reply by waterplea** [reply, marked as answer]
In your stackblitz when I add markForCheck it fixes the issue. As it is it is working as expected, without notifying Angular change detection while doing async operations in OnPush all sorts of different issues might pop up — this is not the correct usage. 

It works part-way because Angular imperatively updates state of the checkbox without notifying change detection, so styles that depend on CSS `:checked` are applied, but some other things rely on bindings that are not updated because nothing has notified the view that change detection must be ran.

In other words, either use default change detection strategy, or use OnPush properly. If your state management calls markForCheck it must be calling it at a wrong time or at a wrong view. If you can reproduce this issue while notifying Angular change detection — please post a stackblitz reproduction.
